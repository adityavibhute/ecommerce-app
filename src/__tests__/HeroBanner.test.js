import { render, screen, fireEvent } from '@testing-library/react';
import HeroBanner from '../components/HeroBanner';
import { MemoryRouter } from 'react-router-dom';

describe('test HeroBanner components', () => {
    it('should test HeroBanner component', () => {
        render(
            <MemoryRouter>
              <HeroBanner />
            </MemoryRouter>
          );
        expect(screen.getByText('Previous')).toBeInTheDocument();
    });
    it('check for the next and prev click', () => {
      const { getByTestId } = render(
        <MemoryRouter>
          <HeroBanner />
        </MemoryRouter>
      );
      const nextButton = getByTestId('next-click');
      const prevButton = getByTestId('previous-click');
      fireEvent.click(nextButton);
      expect(screen.getByText('Next')).toBeInTheDocument();
      fireEvent.click(prevButton);
      expect(screen.getByText('Previous')).toBeInTheDocument();
    })
});
