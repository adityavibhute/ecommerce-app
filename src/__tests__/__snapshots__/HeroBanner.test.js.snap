// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HeroBanner components should test HeroBanner component 1`] = `
<Fragment>
  <div
    className="overflow-hidden relative"
  >
    <div
      className="flex transition-transform ease-out duration-500"
      style={
        Object {
          "transform": "translateX(-0% )",
        }
      }
    >
      <img
        alt="hero-image=./slide-one.jpeg"
        key="./slide-one.jpeg"
        src="slide-one.jpeg"
      />
      <img
        alt="hero-image=./slide-two.jpg"
        key="./slide-two.jpg"
        src="slide-two.jpg"
      />
      <img
        alt="hero-image=./slide-three.jpeg"
        key="./slide-three.jpeg"
        src="slide-three.jpeg"
      />
    </div>
    <button
      className="absolute bottom-0 left-0 top-0 z-[1] flex w-[15%] items-center justify-center border-0 bg-none p-0 text-center text-white opacity-50 transition-opacity duration-150 ease-[cubic-bezier(0.25,0.1,0.25,1.0)] hover:text-white hover:no-underline hover:opacity-90 hover:outline-none focus:text-white focus:no-underline focus:opacity-90 focus:outline-none motion-reduce:transition-none"
      data-te-slide="prev"
      data-te-target="#carouselExampleControls"
      onClick={[Function]}
      type="button"
    >
      <span
        className="inline-block h-8 w-8"
      >
        <svg
          className="h-6 w-6"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5L8.25 12l7.5-7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </span>
      <span
        className="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
      >
        Previous
      </span>
    </button>
    <button
      className="absolute bottom-0 right-0 top-0 z-[1] flex w-[15%] items-center justify-center border-0 bg-none p-0 text-center text-white opacity-50 transition-opacity duration-150 ease-[cubic-bezier(0.25,0.1,0.25,1.0)] hover:text-white hover:no-underline hover:opacity-90 hover:outline-none focus:text-white focus:no-underline focus:opacity-90 focus:outline-none motion-reduce:transition-none"
      data-te-slide="next"
      data-te-target="#carouselExampleControls"
      onClick={[Function]}
      type="button"
    >
      <span
        className="inline-block h-8 w-8"
      >
        <svg
          className="h-6 w-6"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </span>
      <span
        className="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
      >
        Next
      </span>
    </button>
    <div
      className="absolute bottom-4 right-0 left-0"
    >
      <div
        className="flex items-center justify-center gap-2"
      >
        <div
          className="
                        transition-all w-3 h-3 bg-white rounded-full 
                        p-2"
          key="./slide-one.jpeg"
        />
        <div
          className="
                        transition-all w-3 h-3 bg-white rounded-full 
                        bg-opacity-50"
          key="./slide-two.jpg"
        />
        <div
          className="
                        transition-all w-3 h-3 bg-white rounded-full 
                        bg-opacity-50"
          key="./slide-three.jpeg"
        />
      </div>
    </div>
  </div>
  <Card />
</Fragment>
`;
